lib/AST/Type.cpp

```diff
@@ -134,7 +134,8 @@ bool TypeBase::isUninhabited() {
   // Empty enum declarations are uninhabited
   if (auto nominalDecl = getAnyNominal())
     if (auto enumDecl = dyn_cast<EnumDecl>(nominalDecl))
-      if (enumDecl->getAllElements().empty())
+      // Objective-c enums may yet be inhabited.
+      if (enumDecl->getAllElements().empty() && !enumDecl->isObjC())
```

Comment:
This should be restricted to `enumDecl->isObjC() && enumDecl->hasClangNode()` because we hold Swift-defined ObjC enums to a higher standard.